Add methods to bounce pane view with a completion block.

fcanas opened the pull request:
Sometimes it's necessary to perform actions when a bounce is completed. Since the animation is dynamic, it's not appropriate to accomplish this with timers.

Please review the approach I took hooking into the dynamic animator completion block. I think it is right, but I do not fully understand the possible internal states this object may have.

fcanas committed:
Add methods to bounce pane view with a completion block.

fcanas commented:
I'm open to any other way to run code after the bounce action completes. I didn't see any way to do that with the existing APIs. And since there's no existing structure for delegation, I thought this might be the best way.

erichoracek commented:
The internal implementation looks good to me—that's what the internal dynamics completion block is for! :smile: 

As for the API, I'd suggest a few changes. I totally agree that you'll want to do something after you bounce the pane open—in fact I'd go as far as to say that you'll almost always want to do something afterwards. I'd love to keep the number of methods down and as such would recommend just replacing `bouncePaneOpen` method with `bouncePaneOpen:allowUserInterruption:completion:` and the same for the equivalent "directional" bounce method (currently `bouncePaneOpenInDirection:`). I think the `allowUserInterruption` `BOOL` will not only easily enable preventing user interaction during the animation but to inform the consumers of the API that the user would be able to interrupt this animation in the first place. A good example of how to implement this using the internal methods is in implementation of the `setPaneState:inDirection:animated:allowUserInterruption:completion:` method.

Finally, I noticed you added question marks between the `@param` and the `@see`—those newlines are there to make them easier to read (I find that it helps, even though I know it's not the standard).

Let me know what you think.

fcanas commented:
I'll move the implementation of the bounce feature in that direction. The user interruption parameter will certainly be useful. Set it as interruptible for behavior like Apple's camera on the lock screen, and uninterruptible for UX hinting.

I'll move the interface in that direction...

fcanas committed:
Add parameter to dis/allow user interaction during a bounce.
Also changed method signature in example code, though did not expose switch for user interaction flag in the example app.

fcanas commented:
Updated PR. Is this what you intended for the interface?

A switch to the example code could probably be added to demonstrate the option for user interaction.

fcanas commented:
It might be important to note that an API change that's not backwards-compatible generally warrants a full version bump. It might be prudent to not remove the existing methods as I have done.

http://semver.org

fcanas committed:
Re-add previous methods for bouncing the pane view.

fcanas commented:
Should be good to go as a minor release wrt semantic versioning.

Not that this warrants a release if you don't want to make one right now. Just that _I_ didn't break anything. :-)
